packages/devtools-source-map/src/utils/fetchSourceMap.js

```diff
@@ -9,95 +9,41 @@ const { getSourceMap, setSourceMap } = require("./sourceMapRequests");
 const { WasmRemap } = require("./wasmRemap");
 const { SourceMapConsumer } = require("source-map");
 
-const path = require("./path");
-
-const URL_ISH = new RegExp("^[a-z]+:/");
-
 import type {
-  Location,
   Source,
-  SourceScope,
-  MappedScopeBindings
 } from "debugger-html";
 
-/**
- * Sets the source map's sourceRoot to be relative to the source map url.
- * @memberof utils/source-map-worker
- * @static
- */
-function _setSourceMapRoot(
-  sourceMap: Object,
-  absSourceMapURL: string,
-  source: Source
-) {
-  // No need to do this fiddling if we won't be fetching any sources
-  // over the wire.  However, we do still want to if any of the source
-  // URLs are relative.  What's difficult is that we want to pretend
-  // that some non-URLs, like "webpack:/whatever", are actually URLs.
-  if (sourceMap.hasContentsOfAllSources()) {
-    const allURLsAreAbsolute = sourceMap.sources.every(sourceName => {
-      return URL_ISH.test(sourceName);
-    });
-    if (allURLsAreAbsolute) {
-      return;
-    }
-  }
-
-  // If it's already a URL, just leave it alone.
-  if (!path.isURL(sourceMap.sourceRoot)) {
-    // In the odd case where the sourceMap is a data: URL and it does
-    // not contain the full sources, fall back to using the source's
-    // URL, if possible.
-    let parsedSourceMapURL = new URL(absSourceMapURL);
-
-    if (parsedSourceMapURL.protocol === "data:" && source.url) {
-      parsedSourceMapURL = new URL(source.url);
-    }
-
-    parsedSourceMapURL.pathname = path.dirname(parsedSourceMapURL.pathname);
-    // Gecko's URL constructor can fail if the empty string is passed
-    // as the first parameter.  See
-    // https://github.com/devtools-html/debugger.html/issues/3816
-    if (sourceMap.sourceRoot) {
-      parsedSourceMapURL = new URL(sourceMap.sourceRoot, parsedSourceMapURL);
-    }
-    sourceMap.sourceRoot = parsedSourceMapURL.toString();
-  }
-}
-
 function _resolveSourceMapURL(source: Source) {
   const { url = "", sourceMapURL = "" } = source;
-  if (path.isURL(sourceMapURL) || url == "") {
-    // If it's already a full URL or the source doesn't have a URL,
-    // don't resolve anything.
-    return sourceMapURL;
+
+  if (!url) {
+    // If the source doesn't have a URL, don't resolve anything.
+    return { sourceMapURL, baseURL: sourceMapURL };
   }
 
-  if (path.isAbsolute(sourceMapURL)) {
-    // If it's an absolute path, it should be resolved relative to the
-    // host of the source.
-    const { protocol = "", host = "" } = new URL(url);
-    return `${protocol}//${host}${sourceMapURL}`;
+  const resolvedURL = new URL(sourceMapURL, url);
```

Comment:
Should this be wrapped in a try/catch block ? `new URL` will throw if `url` is not a url (but maybe this can't happen here ?)

Reply:
I see now this can be simplified a little.

However, if this doesn't resolve to a URL one way or another, then I think we want to fail -- because it means the caller passed in something wrong.